### PR TITLE
[CHORE] Simplify/remove mutation from updateTaskDefinitionImage

### PR DESCRIPTION
**Pull request by maschwenk**
I was trying to debug an issue and noticed this code was a little confusing and relied on mutation. WDYT @coen-hyde

### CODE REVIEW

**Comment by maschwenk** [comment]
give me a sec, trying to corrale these renames

**Comment by maschwenk** [comment]
Coen this is the only spec failing, I'm having a little trouble following that spec. Any idea?
```
 1) ECS Service Image Updater
       Wrap up
         should do it all more good:
     Error: expected undefined to equal 'arn:updated'
```

File: index.js [hunk 1]
```diff
@@ -133,25 +140,10 @@ Object.assign(updater, {
   },
 
   updateTaskDefinitionImage(taskDefinition, containerNames, image) {
```
Comment (maschwenk) [inline index.js]: now that this is not mutating, maybe a good idea to rename to something that sounds non mutatey, or just move it up into the function up above

File: index.js [hunk 2]
```diff
@@ -133,25 +140,10 @@ Object.assign(updater, {
...
-      'containerDefinitions',
-      'family',
-      'networkMode',
-      'placementConstraints',
-      'taskRoleArn',
-      'volumes'
-    ]);
+    return taskDefinition.containerDefinitions.map(
+      def =>
+        [].concat(containerNames).includes(def.name) ? { ...def, image } : def
```
Comment (maschwenk) [inline index.js]: `[].concat(containerNames)` array-ifies the arguments:
```javascript
[].concat('boop')   // => ['boop']
[].concat(['boop'])   // => ['boop']
```

Comment (coen-hyde) [reply index.js]: that's a nice hack

**Comment by coen-hyde** [comment]
I like your changes. Though the `updater` function is getting rather fat now. Lets split out the generation of the new task definition into a `generateTaskDefinition` function which will take the same arguments as the `updateTaskDefinitionImage` but return a whole task definition rather than a container list. The `updateTaskDefinitionImage` function can go away with the container map in `generateTaskDefinition`.

**Comment by maschwenk** [comment]
Spring cleaning my old PR's and I don't really care to follow up on this unless you do @coen-hyde

**Pull request closed without merging**